### PR TITLE
Update zh-Hans homepage

### PR DESCRIPTION
**Pull request by Cierra-Runis**

### CODE REVIEW

**Comment by rrousselGit** [comment]
Hello! I appreciate the work, but it's a bit difficult to review as is.  
There are some unrelated changes in there.

Could you separate the zh-Hans changes from the rest?

File: website/src/components/CodeSnippet/index.tsx [hunk 1]
```diff
@@ -32,7 +32,11 @@ interface CodeSnippetProps {
   snippet: string;
 }
 
-export const CodeSnippet: React.FC<CodeSnippetProps> = ({ snippet, title, ...other }) => {
+export const CodeSnippet: React.FC<CodeSnippetProps> = ({
+  snippet,
+  title,
+  ...other
+}) => {
```
Comment (rrousselGit) [inline website/src/components/CodeSnippet/index.tsx:39]: What's the purpose of these changes?

File: website/src/pages/index.tsx [hunk 2]
```diff
@@ -9,21 +9,66 @@ import { highlights } from "../data/highlights";
...
+    key: "homepage.features_logger_check",
+    value: <Translate>Logging</Translate>,
+  },
+  {
+    key: "homepage.features_socket_check",
+    value: <Translate>Websocket support</Translate>,
+  },
+  {
+    key: "homepage.features_graph_check",
+    value: <Translate>Documentation generator</Translate>,
```
[the remark this answers is not in the document]
Comment (rrousselGit) [reply website/src/pages/index.tsx:70]: But fair enough then.

File: website/src/pages/index.tsx [hunk 3]
```diff
@@ -47,17 +92,15 @@ export default function Home() {
...
           <h2>
-            <Translate
-              id="homepage.features_title"
-            >
-              {`Features`}
-            </Translate>
+            <Translate id="homepage.features_title">{`Features`}</Translate>
           </h2>
+          {
+            // TIPS: TEXT LABELS MUST BE STATIC
```
Comment (rrousselGit) [inline website/src/pages/index.tsx:98]: Could you move this to the `checkBoxes` variable instead?

Comment (Cierra-Runis) [reply website/src/pages/index.tsx:98]: I will try it :)

**Comment by Cierra-Runis** [comment]
> Hello! I appreciate the work, but it's a bit difficult to review as is. There are some unrelated changes in there.
> 
> Could you separate the zh-Hans changes from the rest?

I am sorry to put you through a little bit of work😢, but I don't know how to separate the zh-Hans changes now 😞.


I have just used the PR and branch of github, and I am not very familiar with how to use them. I just asked my classmate, and he suggested that I open more PR in the future instead single PR.

All the changes I made are for translation, don't worry, you've seen all the reasonable changes I made under the docs folder.

**Comment by rrousselGit** [comment]
It's alright, I guess I'll deal with it. 

But I still don't understand why the snippet component had to be changed

**Comment by rrousselGit** [comment]
@Cierra-Runis So, could you explain why the snippet component was changed? :)

**Comment by Cierra-Runis** [comment]
> @Cierra-Runis So, could you explain why the snippet component was changed? :) 

Oh, it seems that you didn't received the reply at https://github.com/rrousselGit/riverpod/pull/2539#discussion_r1192160203

**Comment by Cierra-Runis** [comment]
> @Cierra-Runis So, could you explain why the snippet component was changed? :) 

At website/docs/concepts/reading.mdx, I add index before title.



## Before: showing title 1 and 2 with `flutter_hooks` closed.

https://github.com/rrousselGit/riverpod/assets/29329988/8551607e-c46e-4a30-8fe6-156c1821f14f

showing title 1~5 with `flutter_hhooks` enabled, but code at 1 and 3, 2
and 4 is repeated.

https://github.com/rrousselGit/riverpod/assets/29329988/f3985974-fe5f-45e6-b304-58a5c8ebc55d

## After: make it doesn't repeated.

**Comment by rrousselGit** [comment]
Oh, I see! Thanks!

To begin with, the duplication in the approach should be removed. It shouldn't be that both ConsumerWidget and ConsumerHookWidget are present in the same file.
Instead enabling/disabling flutter_hooks should enable/disable the associated paragraphs.

File: website/docs/concepts/reading.mdx [hunk 4]
```diff
@@ -60,7 +61,7 @@ solutions to obtain one from widgets.
 
 <ConditionalSnippet>
 
-#### Extending ConsumerWidget instead of StatelessWidget
+#### 1. Extending ConsumerWidget instead of StatelessWidget
```
Comment (rrousselGit) [inline website/docs/concepts/reading.mdx:64]: Could you remove those numbers? That's not a practice the rest of the website uses.

**Comment by rrousselGit** [comment]
:wave:

Do you mind fixing the conflicts? It's a bit hard for me to do so since I don't understand Chinese :P

Otherwise LGTM

**Comment by Cierra-Runis** [comment]
It's too messy, I will reopen PRs.

**Comment by cn-tre-lfp** [comment]
there is a error in chinese doc.
[https://docs-v2.riverpod.dev/zh-hans/docs/riverpod_for_provider_users](url) 
look at '缓存和**小会**一些有状态的对象', it should be '缓存和**释放**一些有状态的对象'
@Cierra-Runis
